specification/keyvault/data-plane/TypeSpec/routes.tsp

```diff
@@ -0,0 +1,278 @@
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-client-generator-core";
+import "@typespec/rest";
+import "./models.tsp";
+
+using Azure.Core;
+using Azure.Core.Traits;
+using Azure.ClientGenerator.Core;
+using TypeSpec.Rest;
+using TypeSpec.Http;
+
+namespace KeyVault;
+
+alias KeyVaultOperations = StandardResourceOperations;
+
+alias KeyVaultListParameters = ListQueryParametersTrait<{
+  /**
+   * Maximum number of results to return in a page. If not specified the service
+   * will return up to 25 results.
+   */
+  @maxValue(25)
+  @minValue(1)
+  @query("maxresults")
+  maxresults?: int32;
+}>;
+
+/**
+ *  The SET operation adds a secret to the Azure Key Vault. If the named secret
+ * already exists, Azure Key Vault creates a new version of that secret. This
+ * operation requires the secrets/set permission.
+ */
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Foundations.Operation is necessary for Key Vault"
+@summary("Sets a secret in a specified key vault.")
+@route("/secrets/{secretName}")
+@put
+op setSecret is Azure.Core.Foundations.Operation<
+  {
+    /**
+     * The name of the secret. The value you provide may be copied globally for the
+     * purpose of running the service. The value provided should not include
+     * personally identifiable or sensitive information.
+     */
+    @pattern("^[0-9a-zA-Z-]+$")
+    @path
+    secretName: string;
```

Comment:
I believe it's just "name" for all SDKs. Should that use a decorator instead? The old swagger uses "secret-name". Same for "secret-version" below.